onnx/common/ir.h

```diff
@@ -767,6 +767,56 @@ struct Node : public Attributes<Node> {
   }
 };
 
+// A class with the same properties as OperatorSetIdProto, but without protobuf
+// overhead, resulting in a simpler and more readable workflow.
+class OpSetID final {
+  private:
+    std::string domain_;
+    int64_t version_;
+
+  public:
+    explicit OpSetID(const ONNX_NAMESPACE::OperatorSetIdProto& proto)
+      :domain_(proto.domain()), version_(proto.version()) {}
+
+    // Default Domain Constructor
+    explicit OpSetID(const int64_t version)
+      :domain_(""), version_(version) {}
+
+    explicit OpSetID(const std::string domain, int64_t version)
+      :domain_(std::move(domain)), version_(version) {}
+
+    std::string toString() const {
+      return "$" + domain_ + "$" + std::to_string(version_);
+    }
+
+    static OpSetID fromString(const std::string& target) {
+      try {
+        std::string new_domain = target.substr(0, target.find("$"));
```

Comment:
what is the format of target? Shall we add it to the comment?
If target starts with "$", this logic does not seem right.

Reply:
More legacy relative to adapters branch